rs/tests/execution/general_execution_tests/nns_shielding.rs

```diff
@@ -141,6 +142,103 @@ pub fn mint_cycles_not_supported_on_application_subnet(env: TestEnv) {
     });
 }
 
+pub fn mint_cycles128_supported_only_on_cycles_minting_canister(env: TestEnv) {
+    let nns_node = env.get_first_healthy_nns_node_snapshot();
+    let specified_id = nns_node.get_last_canister_id_in_allocation_ranges();
+    // Check that 'specified_id' is not 'CYCLES_MINTING_CANISTER_ID'.
+    assert_ne!(specified_id, CYCLES_MINTING_CANISTER_ID.into());
+    let nns_agent = nns_node.build_default_agent();
+    block_on(async move {
+        let not_cmc = UniversalCanister::new_with_cycles(&nns_agent, specified_id, *INITIAL_CYCLES)
```

Comment:
the function `new_with_cycles` takes an "effective" canister id rather than a "specified" canister id:
- effective: any canister id used for routing (can be reused for multiple canister creations),
- specified: the new canister is created with this particular canister id (cannot be reused).

Reply:
I'll rename it. 

Just to clarify, 

>  (can be reused for multiple canister creations)

How? Unless it is the subnet's management canister.